### PR TITLE
update: Add permission check before starting to update

### PR DESCRIPTION
**Pull request by vadmeste**
## Description
For a better user experience, perform permission check before
update as exposed by update library

## Motivation and Context
Better UX

## How to test this PR?
make install && sudo install minio /usr/local/bin/minio; /usr/local/bin/minio server /tmp/xl/{1..4}
mc admin update myminio

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Documentation updated
- [ ] Unit tests added/updated

### CODE REVIEW

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-zoned.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

File: cmd/update.go [hunk 1]
```diff
@@ -537,6 +537,14 @@ func doUpdate(u *url.URL, lrTime time.Time, sha256Sum []byte, releaseInfo string
 		Checksum: sha256Sum,
 	}
 
+	if err := opts.CheckPermissions(); err != nil {
```
Comment (harshavardhana) [inline cmd/update.go:540]:
```suggestion
	if err := opts.CheckPermissions(); err != nil {
```

🍰

**Review by harshavardhana — APPROVED** [review]

File: cmd/update.go (hunk 1 above)
Comment (vadmeste) [reply cmd/update.go:540]: I am not sure what is this suggestion and I am not able to apply it as well.

Comment (harshavardhana) [reply cmd/update.go:540]: @vadmeste its just another way to say "very nice"

**Pull request merged**